java-checks/src/main/java/org/sonar/java/checks/AnonymousClassShouldBeLambdaCheck.java

```diff
@@ -70,20 +75,54 @@ public void visitNewClass(NewClassTree tree) {
     if (classBody != null) {
       TypeTree identifier = tree.identifier();
       if (!useThisIdentifier(classBody) && !enumConstants.contains(identifier) && isSAM(classBody)) {
-        context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        MethodTree method = (MethodTree) classBody.members().get(0);
+        JavaSymbol.TypeJavaSymbol interfaceOwner = (JavaSymbol.TypeJavaSymbol) identifier.symbolType().symbol();
+        if (multipleDefaultMethodsWithSameSignature(((JavaSymbol.MethodJavaSymbol) method.symbol()), interfaceOwner)) {
+          context.reportIssue(this, identifier, "Make this anonymous inner class a lambda" + context.getJavaVersion().java8CompatibilityMessage());
+        }
       }
     }
   }
 
   private static boolean isSAM(ClassTree classBody) {
-    if(hasOnlyOneMethod(classBody.members()) && classBody.symbol().isTypeSymbol()) {
+    if (hasOnlyOneMethod(classBody.members()) && classBody.symbol().isTypeSymbol()) {
       // Verify class body is a subtype of an interface
       JavaSymbol.TypeJavaSymbol symbol = (JavaSymbol.TypeJavaSymbol) classBody.symbol();
-      return symbol.getInterfaces().size() == 1 && symbol.getSuperclass().is("java.lang.Object");
+      return symbol.getInterfaces().size() == 1 &&
+        symbol.getSuperclass().is("java.lang.Object");
     }
     return false;
   }
 
+  private static boolean multipleDefaultMethodsWithSameSignature(JavaSymbol.MethodJavaSymbol methodSymbol, JavaSymbol.TypeJavaSymbol interfaceOwner) {
+    Set<JavaSymbol.MethodJavaSymbol> methods = new HashSet<>();
+    List<TypeJavaSymbol> allInterfacesOfHierarchy = interfaceOwner.superTypes().stream().filter(type -> !type.is("java.lang.Object"))
+      .map(ClassJavaType::getSymbol).collect(Collectors.toList());
+    for (JavaSymbol.TypeJavaSymbol interfaceOfHierarchy : allInterfacesOfHierarchy) {
+      methods.addAll(interfaceOfHierarchy.memberSymbols().stream().filter(Symbol::isMethodSymbol).map(method -> ((JavaSymbol.MethodJavaSymbol) method))
+        .filter(method -> !method.name().equals(methodSymbol.name()))
+        .filter(method -> sameParameters(method, methodSymbol))
+        .collect(Collectors.toList()));
+    }
+    methods.addAll(
+      interfaceOwner.memberSymbols().stream().filter(Symbol::isMethodSymbol).map(method -> ((JavaSymbol.MethodJavaSymbol) method))
+        .filter(method -> sameParameters(method, methodSymbol))
+        .collect(Collectors.toList()));
+    // methods.retainAll(methods.stream().filter(method -> sameParameters(method, methodSymbol)).collect(Collectors.toList()));
+    return checkForDefaultMethodsWithSameSignature(methodSymbol, methods);
+  }
+
+  private static boolean checkForDefaultMethodsWithSameSignature(JavaSymbol.MethodJavaSymbol methodSymbol, Set<JavaSymbol.MethodJavaSymbol> methods) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unused method parameter "methodSymbol". ([squid:S1172](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=1946&issues=AWJI1Iz6uc1UcxX5-Ok3&open=AWJI1Iz6uc1UcxX5-Ok3)

[ISSUE_KEY]: AWJI1Iz6uc1UcxX5-Ok3